File: FlowerShop/src/app/services/order.service.ts
import { ItemOrderDto } from 'src/app/services/cart';
import { OrderStatus, PaymentType, ProductOrderDto } from './order';
import { OrderServerService } from './server/order-server.service';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private orderServerService: OrderServerService) { }

  public sendOrder(deliveryAddress: string, deliveryName: string, email: string, paymentType: PaymentType, phone: string, productItems: ItemOrderDto[], id?: number, orderStatus?: OrderStatus, text?: string) {
    const dto: ProductOrderDto = {
      deliveryAddress,
      deliveryName,
      email,
      paymentType,
      phone,
      productItems,
      id,
      orderStatus,
      text
    };
    return this.orderServerService.post(dto);
  }
}

File: FlowerShop/services/OrderService.cs
using FlowerShop.models.db;
using FlowerShop.models.dto;
using System;
using System.Collections.Generic;

namespace FlowerShop.services
{
    public static class OrderService
    {
        public static Order GetClientOrder(OrderDB databaseOrder, List<CartItemDB> dbCartItems)
        {
            if (databaseOrder is not null)
            {
                return new Order()
                {
                    id = databaseOrder.Id,
                    deliveryAddress = databaseOrder.Address,
                    deliveryName = databaseOrder.Name,
                    email = databaseOrder.Email,
                    orderStatus = databaseOrder.Status,
                    paymentType = databaseOrder.PaymentType,
                    phone = databaseOrder.Phone,
                    text = "",
                    productItems = CartService.GetClientItemsArray(dbCartItems),
                };
            }
            return null;
        }

        public static OrderDB GetDatabaseOrder(Order clientOrder)
        {
            if (clientOrder is not null)
            {
                return new OrderDB()
                {
                    Address = clientOrder.deliveryAddress,
                    Name = clientOrder.deliveryName,
                    Email = clientOrder.email,
                    Status = clientOrder.orderStatus,
                    PaymentType = clientOrder.paymentType,
                    Phone = clientOrder.phone,
                };
            }
            return null;
        }
    }
}
